addon/addon/models/sprite-snapshot-node-builder.ts

```diff
@@ -203,15 +203,20 @@ export class SpriteSnapshotNodeBuilder {
     let freshlyChanged: Set<SpriteModifier> = new Set();
     for (let context of contexts) {
       context.captureSnapshot();
-      let contextDescendants = this.spriteTree.descendantsOf(context);
-      for (let contextDescendant of contextDescendants) {
-        if (contextDescendant instanceof SpriteModifier) {
-          let spriteModifier = contextDescendant as SpriteModifier;
-          spriteModifier.captureSnapshot();
-          // TODO: what about refactoring away checkForChanges and simply treating all leftover sprites in the SpriteTree as KeptSprites
-          if (checkForChanges(spriteModifier, context)) {
-            freshlyChanged.add(spriteModifier);
-          }
+      let contextNode = this.spriteTree.nodesByElement.get(context.element);
+      let contextChildren: SpriteModifier[] = [...(contextNode?.children ?? [])]
```

Comment:
We don't really need to destructure here, but is it good practice to?

Reply:
We need the destructure because `children` is a `Set` so no `.map()` available.

And yes, I'd like to refactor the SpriteTree to have util methods that are closer to what we need now. I think we're not using the existing descendant related methods anymore other than for the `filterToContext` utility, which could do it internally.

Reply:
Oops, yes - I forget that it is a set